src/pages/Post.js

```diff
@@ -656,81 +711,61 @@ class PostPage extends React.Component {
-                        Save Cropped Image
-                      </button>
-                    {/*croppedImageUrl && (
-                      <img alt="Crop" style={{ maxWidth: "100%" }} src={croppedImageUrl} />
-                    )*/}
+                    <div style={{margin: "20px 0px 20px 0px", float: "center", verticalAlign: "middle", clear: "left" }}>
+                        <button id="buttonOpenCrop" className="buttonOpenCrop" onClick={this.openModal} style={{
+                            margin: "16px 0px 0px 0px",
+                            width: 115,
+                            height: 30,
+                            boxShadow: "0 2px 4px 0 rgba(0, 0, 0, 0.5)",
+                            border: "solid 0 #979797",
+                            backgroundColor: "#2175cb",
+                            fontFamily: "HelveticaNeue-Bold",
+                            fontWeight: 500,
+                            fontSize: 14,
+                            color: "#ffffff",
+                            display: "none"
+                          }}>
+                            Crop Image
+                        </button>
+                    </div>
                   </div>
+                  
+                  <Modal
+                    isOpen={this.state.modalIsOpen}
+                    onAfterOpen={this.afterOpenModal}
+                    onRequestClose={this.closeModal}
+                    style={this.state.modalStyle}
+                    contentLabel="Cropping Modal">
+                    <center>
+                    <div id="cropping" style={{margin: "16px 40px 0px 40px"}}>
+                      {src && (
+                        <ReactCrop
+                          src={src}
+                          crop={crop}
+                          onImageLoaded={this.onImageLoaded}
+                          onComplete={this.onCropComplete}
+                          onChange={this.onCropChange}
+                        />
+                      )}
+                      <div style={{margin: "20px 0px 20px 0px", float: "center", verticalAlign: "middle", clear: "left" }}>
+                          <button id="buttonCrop" className="buttonCrop" style={{
+                              margin: "16px 0px 0px 0px",
+                              width: 300,
+                              height: 35,
+                              boxShadow: "0 2px 4px 0 rgba(0, 0, 0, 0.5)",
+                              border: "solid 0 #979797",
+                              backgroundColor: "#2175cb",
+                              fontFamily: "HelveticaNeue-Bold",
+                              fontWeight: 500,
+                              fontSize: 18,
+                              color: "#ffffff"
+                            }}>
+                              Save Cropped Image
+                          </button>
+                      </div>
+                    </div>
+                    </center>
+                  </Modal>
```

Comment:
Let's refactor the model into its own subcomponent. We can pass all the necessary functions into it as props.